src/block/TallGrass.php

```diff
@@ -25,14 +25,52 @@
 
 use pocketmine\block\utils\StaticSupportTrait;
 use pocketmine\block\utils\TallGrassTrait;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
 use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
 
 class TallGrass extends Flowable{
 	use TallGrassTrait;
 	use StaticSupportTrait;
 
+	/** @var \Closure() : DoublePlant */
+	private \Closure $doublePlantVariant;
+
+	public function __construct(BlockIdentifier $idInfo, string $name, BlockTypeInfo $typeInfo, \Closure $doublePlantVariant){
+		parent::__construct($idInfo, $name, $typeInfo);
+		$this->doublePlantVariant = $doublePlantVariant;
+	}
+
 	private function canBeSupportedAt(Block $block) : bool{
 		$supportBlock = $block->getSide(Facing::DOWN);
 		return $supportBlock->hasTypeTag(BlockTypeTags::DIRT) || $supportBlock->hasTypeTag(BlockTypeTags::MUD);
 	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($this->getSide(Facing::UP)->getTypeId() !== BlockTypeIds::AIR){
+			return false;
+		}
+
+		$world = $this->position->getWorld();
+		if($item instanceof Fertilizer){
+			$doubleVariant = $this->getDoublePlantVariant();
+
+			$bottom = (clone $doubleVariant)->setTop(false);
+			$top = (clone $doubleVariant)->setTop(true);
+
+			$world->setBlock($this->position, $bottom);
+			$world->setBlock($this->position->getSide(Facing::UP), $top);
+
+			$item->pop();
+			return true;
+		}
+
+		return false;
+	}
+
+	public function getDoublePlantVariant() : DoublePlant{
```

Comment:
Better to keep this private since it's only used for internal purposes right now.

If we want to make it public later on we can do, but API additions need to be thought about more than other types of changes to make sure we don't add stuff we're gonna regret later.